python/pyrenn.py

```diff
@@ -729,15 +730,23 @@ def train_LM(P,Y,net,k_max=100,E_stop=1e-10,dampfac=3.0,dampconst=10.0,\
 			
 			net['w'] = w + w_delta #new weight vector
 			
-			Enew = calc_error(net,data) #calculate new Error E
-			
-			if Enew<E:
+			Enew = calc_error(net,data) #calculate new Error E			
+			if Enew<E and abs(E-Enew)>=min_E_step:
 			#Optimization Step successful!
-				dampfac= dampfac/dampconst #adapt scale factor
+				#if E-Enew<=1e-09:
+				dampfac= dampfac/dampconst#adapt scale factor
 				break #go to next iteration
 			else:
-			#Optimization Step NOT successful!
+			#Optimization Step NOT successful!\
 				dampfac = dampfac*dampconst#adapt scale factor
+				if abs(E-Enew)<=min_E_step:
+					if verbose:
+						print('E-Enew<=min_E_step Encountered!!')
+					early=early+1
+					if early>=5.0:
```

Comment:
I think yo can write this with just one "if verbose" statement

Reply:
I agree.